packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1576,8 +1576,8 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
 
     // Validate against instance type restrictions, per
     // https://docs.aws.amazon.com/opensearch-service/latest/developerguide/supported-instance-types.html
-    if (isSomeInstanceType('i3', 'r6gd', 'i4g', 'im4gn') && ebsEnabled) {
-      throw new Error('I3, R6GD, I4G, and IM4GN instance types do not support EBS storage volumes.');
+    if (isSomeInstanceType('i3', 'r6gd', 'i4g', 'i4i', 'im4gn', 'r7gd') && ebsEnabled) {
```

Comment:
It would be nice to refactor these instance type lists and store them in a named constant, to improve readability and maintenance. You could also use their human-friendlier `InstanceClass` names, something like:

```ts
private static readonly UNSUPPORTED_EBS = [
  ec2.InstanceClass.IO3,
  ec2.InstanceClass.MEMORY6_GRAVITON2_NVME_DRIVE,
  // ...
];

// ...
if(isSomeInstanceType(...UNSUPPORTED_EBS) && ebsEnabled) throw
```

You could also generate the error message using this list, without having to update them both manually:

```ts
function formatInstanceTypesList(instanceTypes: string[]): string {
  return instanceTypes.map((type) => type.toUpperCase()).join(', ').replace(/, ([^,]*)$/, ' and $1');
}

throw new Error(`${formatInstanceTypesList(UNSUPPORTED_EBS)} instance types do not support EBS storage volumes.`);
```

Reply:
Thank you. I took your suggestions into account and tried refactoring the relevant parts for this fix.
As for unrelated parts, I plan to refactor them in a separate PR.